ios/browser/api/brave_rewards/brave_rewards_api.mm

```diff
@@ -226,11 +222,13 @@ - (void)dealloc {
   return options;
 }
 
-- (void)postEngineTask:(void (^)(brave_rewards::internal::RewardsEngine*))task {
-  _engineTaskRunner->PostTask(FROM_HERE, base::BindOnce(^{
-                                CHECK(self->_engine != nullptr);
-                                task(self->_engine.get());
-                              }));
+- (void)postTask:(void (^)())task {
+  auto __weak weakSelf = self;
+  _taskRunner->PostTask(FROM_HERE, base::BindOnce(^{
+                          if (weakSelf) {
+                            task();
+                          }
+                        }));
```

Comment:
I think there's some confusion here around weak/strong, if we want to avoid weak/strong dance in all posted tasks you need to pass in the `weakSelf` to the closure then use that the same way we were using `engine`

e.g.

```objc
- (void)postTask:(void (^)(BraveRewardsAPI*))task {
  auto __weak weakSelf = self;
  _taskRunner->PostTask(FROM_HERE, base::BindOnce(^{
    if (weakSelf) { 
      task(weakSelf) 
    }
  });
}
```
and
```
[self postTask:^(BraveRewardsAPI* api){
  api-> _rewardsEngine->Initialize(...)
}];
```

otherwise we're just strongly capturing self in all of the posted tasks anyways and `weakSelf` will never be nil

Reply:
Gotcha - I didn't quite understand how reference capturing works in Obj-C.

Reply:
Fixed (I think!)